### PR TITLE
Add sampling capability for int fields

### PR DESCRIPTION
**Pull request by moprak-nrel**
## Summary

Adding an int field sampler in response to #1191. This PR generalizes some of the existing infrastructure for sampling fields to also enable sampling of int fields. There is now a new input parameter `int_fields` in addition to `fields` and `derived_fields`. 

## Pull request type

Please check the type of change introduced:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [x] new unit-test(s)
- [x] new regression test(s)
- [x] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [x] on CPU 

## Additional background

Issue Number:

### CODE REVIEW

**Comment by moprak-nrel** [comment]
Still pending tests/docs updates. 

Making `sample_field` and `interpolate_fields` work with both `Field` and `IntField` seemed like the cleanest solution for this PR to me. But.. I'm not thrilled about making `sample_field` a public member function, so open to suggestions on a better way if possible.

File: amr-wind/utilities/sampling/Sampling.cpp [hunk 1]
```diff
@@ -65,6 +73,26 @@ void Sampling::initialize()
...
     }
 
+    // Process field information
+    m_nicomp = 0;
+    for (const auto& fname : int_field_names) {
+        if (!repo.field_exists(fname)) {
+            amrex::Print()
+                << "WARNING: Sampling: Non-existent int_field requested: "
+                << fname
+                << ". This is a mistake or the requested  int_field is a "
```
Comment (marchdf) [inline amr-wind/utilities/sampling/Sampling.cpp:83]: extra space before int?

**Comment by marchdf** [comment]
I have no qualms ;) good work

File: amr-wind/utilities/sampling/Sampling.cpp (hunk 1 above)
Comment (moprak-nrel) [reply amr-wind/utilities/sampling/Sampling.cpp]: sharp eye! thanks for catching that, fixed

**Review by marchdf — APPROVED** [review]

**Pull request merged**